src/uct/ib/ud/base/ud_iface.c

```diff
@@ -137,30 +162,50 @@ uct_ud_ep_t *uct_ud_iface_cep_get_ep(uct_ud_iface_t *iface,
     }
 
     conn_match = ucs_conn_match_get_elem(&iface->conn_match_ctx, peer_address,
-                                         conn_sn, queue_type, is_private);
+                                         conn_sn, queue_type, is_rx_only);
     if (conn_match == NULL) {
         return NULL;
     }
 
     ep = ucs_container_of(conn_match, uct_ud_ep_t, conn_match);
     ucs_assert(ep->flags & UCT_UD_EP_FLAG_ON_CEP);
 
-    if (is_private) {
+    if (is_rx_only) {
+        ucs_assertv((ep->flags & (UCT_UD_EP_FLAG_TX | UCT_UD_EP_FLAG_RX)) ==
+                            UCT_UD_EP_FLAG_RX,
+                    "ep %p: flags 0x%x", ep, ep->flags);
         ep->flags &= ~UCT_UD_EP_FLAG_ON_CEP;
+    } else {
+        ucs_assertv(ep->flags & (UCT_UD_EP_FLAG_TX | UCT_UD_EP_FLAG_RX),
+                    "ep %p: flags 0x%x", ep, ep->flags);
     }
 
     return ep;
 }
 
+int uct_ud_iface_cep_has_ep(uct_ud_ep_t *ep)
+{
+    int res = !!(ep->flags & (UCT_UD_EP_FLAG_TX | UCT_UD_EP_FLAG_RX)) &&
```

Comment:
to have `0` or `1` instead of the result of `ep->flags & (UCT_UD_EP_FLAG_TX | UCT_UD_EP_FLAG_RX)` which could be `UCT_UD_EP_FLAG_TX ` or `UCT_UD_EP_FLAG_RX`  or both.